src/main.cpp

```diff
@@ -433,30 +329,168 @@ BOOL __stdcall AutoDetectCallback (GUID*    lpGUID,
     return TRUE;
 }
 
+
+MM2RawFnHook<WNDPROC> gfxWindowProc(NULL, NULL, 0x04A88F0);
+
 class gfxHandler
 {
 public:
-    static bool gfxAutoDetect(bool *success)
+    static void setRes(int width, int height, int cdepth, int zdepth, bool detectArgs)
+    {
+        LogFile::WriteLine("Additional graphics params enabled.");
+
+        $setRes(width, height, cdepth, zdepth, datArgParser::Get("gfxArgs") ? true : detectArgs);
+    }
+
+    static LRESULT APIENTRY WndProcNew(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
     {
-        if (datArgParser::Get("noautodetect"))
+        switch (uMsg)
         {
-            LogFile::WriteLine("Hooking AutoDetect...");
-            
-            // Hook into the original AutoDetect and replace it with our own version
-            InstallGameCallback("AutoDetectCallback", gameVersion, &AutoDetectCallback, HOOK_JMP,
+            case WM_KEYUP:
+            case WM_SYSKEYUP:
             {
-                { NULL, NULL, 0x4AC030 },
-            });
-        }
+                if (HandleKeyPress(wParam))
+                    return 0;
+            } break;
 
-        return $gfxAutoDetect(success);
-    };
+            case WM_ACTIVATEAPP:
+            {
+                if (wParam == FALSE)
+                {
+                    if (datArgParser::Get("nopause"))
+                    {
+                        return 0;
+                    }
+                }
+            } break;
+        }
+        return gfxWindowProc(hWnd, uMsg, wParam, lParam);
+    }
 
-    static void setRes(int width, int height, int cdepth, int zdepth, bool detectArgs)
+    static void __cdecl gfxWindowCreate(LPCSTR lpWindowName)
     {
-        LogFile::WriteLine("Additional graphics params enabled.");
+        MM2PtrHook<LPCSTR> lpWindowTitle(NULL, NULL, 0x68311C);
+
+        if (lpWindowTitle)
+        {
+            lpWindowName = lpWindowTitle;
+        }
+
+        MM2PtrHook<HWND>    hWndMain(NULL, NULL, 0x6830B8);
```

Comment:
All of these PtrHook's should be moved out of the function, just to keep things neat and organized.